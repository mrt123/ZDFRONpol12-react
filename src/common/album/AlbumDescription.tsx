import AlbumDescriptionEntry from "./AlbumDescriptionEntry";

const albumDescriptionStyles = {
  margin: "8px",
};

export interface AlbumDescriptionProps {
  title: string;
  artist: string;
  price?: number;
}

const AlbumDescription = ({ title, artist, price }: AlbumDescriptionProps) => {
  const priceToDisplay = price || "N/A";

  return (
    <div style={albumDescriptionStyles}>
      <AlbumDescriptionEntry value={title} />
      <AlbumDescriptionEntry value={artist} />
      <AlbumDescriptionEntry value={priceToDisplay} />
    </div>
  );
};

export default AlbumDescription;
