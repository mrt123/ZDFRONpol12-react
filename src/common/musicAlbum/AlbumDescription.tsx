import { daysAgo } from "../dates";
import AlbumDescriptionEntry from "./AlbumDescriptionEntry";

const albumDescriptionStyles = {
  margin: "8px",
};

export interface AlbumDescriptionProps {
  title: string;
  artist: string;
  price?: number;
  releaseDate?: Date;
}

const AlbumDescription = ({
  title,
  artist,
  price,
  releaseDate,
}: AlbumDescriptionProps) => {
  let priceToDisplay;
  if (price) priceToDisplay = price;
  else priceToDisplay = "N/A";

  const releaseDateToDisplay = releaseDate ? daysAgo(releaseDate) : null;
  const daysSinceRelease =
    releaseDateToDisplay === null
      ? null
      : `released ${releaseDateToDisplay} days ago`;

  return (
    <div style={albumDescriptionStyles}>
      <AlbumDescriptionEntry value={title} />
      <AlbumDescriptionEntry value={artist} />
      <AlbumDescriptionEntry value={priceToDisplay} />
      <AlbumDescriptionEntry value={daysSinceRelease} />
    </div>
  );
};

export default AlbumDescription;
