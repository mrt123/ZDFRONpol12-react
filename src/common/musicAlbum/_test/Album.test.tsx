import { render } from "@testing-library/react";
import Album from "../Album";
import { AlbumDescriptionProps } from "../AlbumDescription";

test("Album test", () => {
  const albumDescription: AlbumDescriptionProps = {
    title: "artysta",
    artist: "Kamil WG",
  };
  const { container } = render(
    <Album
      number={0}
      coverImageUrl="https://d1csarkz8obe9u.cloudfront.net/posterpreviews/rap-trap-mixtape-cover-art-template-design-42d0a5596257ec487408e1039eac95d7_screen.jpg?ts=1580857100"
      description={albumDescription}
    />
  );

  expect(container).toMatchSnapshot();
});
