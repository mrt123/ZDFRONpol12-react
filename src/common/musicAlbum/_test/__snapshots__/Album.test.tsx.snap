// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album test 1`] = `
<div>
  <div
    data-testid="Album"
    style="display: flex; margin: 20px; font-size: 12px; background: rgb(53, 58, 69);"
  >
    <div
      style="padding: 5px; background: black;"
    >
      0
    </div>
    <div
      style="width: 100px; height: 100px; background: blue; background-image: url(https://d1csarkz8obe9u.cloudfront.net/posterpreviews/rap-trap-mixtape-cover-art-template-design-42d0a5596257ec487408e1039eac95d7_screen.jpg?ts=1580857100); background-size: contain;"
    >
      Â 
    </div>
    <div
      style="margin: 8px;"
    >
      <div
        style="margin: 5px; font-size: 11px;"
      >
        artysta
      </div>
      <div
        style="margin: 5px; font-size: 11px;"
      >
        Kamil WG
      </div>
      <div
        style="margin: 5px; font-size: 11px;"
      >
        N/A
      </div>
      <div
        style="margin: 5px; font-size: 11px;"
      >
        released: unknown
      </div>
    </div>
  </div>
</div>
`;
